### PR TITLE
simulation decoders for kava modules

### PR DESCRIPTION
**Pull request by fedekunze**
- [x] Implement store decoder for all kava modules
- [x] Tests

### CODE REVIEW

File: x/cdp/simulation/decoder.go [hunk 1]
```diff
@@ -1,12 +1,60 @@
...
-	// TODO implement this
-	return ""
+	switch {
+	case bytes.Equal(kvA.Key[:1], types.CdpIDKeyPrefix):
+		var cdpIDsA, cdpIDsB []uint64
+		cdc.MustUnmarshalBinaryLengthPrefixed(kvA.Value, &cdpIDsA)
+		cdc.MustUnmarshalBinaryLengthPrefixed(kvB.Value, &cdpIDsB)
+		return fmt.Sprintf("%v\n%v", cdpIDsA, cdpIDsB)
+
+	case bytes.Equal(kvA.Key[:1], types.CdpIDKey),
```
Comment (karzak) [inline x/cdp/simulation/decoder.go:25]: Is there a reason you didn't include `types.CdpKeyPrefix`?

File: x/bep3/simulation/decoder.go [hunk 2]
```diff
@@ -1,12 +1,37 @@
...
+	"github.com/kava-labs/kava/x/bep3/types"
 )
 
 // DecodeStore unmarshals the KVPair's Value to the module's corresponding type
 func DecodeStore(cdc *codec.Codec, kvA, kvB cmn.KVPair) string {
-	// TODO implement this
-	return ""
+	switch {
+	case bytes.Equal(kvA.Key[:1], types.AtomicSwapKeyPrefix):
+		var swapA, swapB *types.AtomicSwap
```
Comment (rhuairahrighairidh) [inline x/bep3/simulation/decoder.go:17]: Is this meant to be just `types.AtomicSwap`, not a pointer?

File: x/pricefeed/simulation/decoder.go [hunk 3]
```diff
@@ -1,12 +1,31 @@
...
-	return ""
+	switch {
+	case bytes.Contains(kvA.Key, []byte(types.CurrentPricePrefix)):
+		var priceA, priceB types.CurrentPrice
+		cdc.MustUnmarshalBinaryBare(kvA.Value, &priceA)
+		cdc.MustUnmarshalBinaryBare(kvB.Value, &priceB)
+		return fmt.Sprintf("%s\n%s", priceA, priceB)
+
+	case bytes.Contains(kvA.Key, []byte(types.RawPriceFeedPrefix)):
+		var postedPriceA, postedPriceB types.PostedPrice
```
Comment (rhuairahrighairidh) [inline x/pricefeed/simulation/decoder.go:23]: Looks like the type should be a slice here?
Values stored under keys starting with `RawPriceFeedPrefix` use type `[]types.PostedPrice`.

File: x/cdp/simulation/decoder.go (hunk 1 above)
Comment (fedekunze) [reply x/cdp/simulation/decoder.go:25]: maybe I missed it? I'll double check

File: x/bep3/simulation/decoder.go (hunk 2 above)
Comment (fedekunze) [reply x/bep3/simulation/decoder.go]: EDIT: you were right. Idk why I thought that this was a pointer. Sorry about that

**Pull request merged**